### PR TITLE
[TS-403]. Add touchRule implementation.

### PR DESCRIPTION
**Pull request by Ilya-Loginov-Exactpro**

### CODE REVIEW

File: src/main/java/com/exactpro/th2/sim/impl/Simulator.java [hunk 1]
```diff
@@ -163,6 +165,22 @@ public void getRulesInfo(Empty request, StreamObserver<RulesInfo> responseObserv
         responseObserver.onCompleted();
     }
 
+    @Override
+    public void touchRule(TouchRequest request, StreamObserver<Empty> responseObserver) {
+        SimulatorRuleInfo ruleInfo = ruleIds.get(request.getId().getId());
+        if (ruleInfo == null || ruleInfo.getRule() == null) {
```
Comment (Nikita-Smirnov-Exactpro) [inline src/main/java/com/exactpro/th2/sim/impl/Simulator.java:171]: Let's modify ruleInfo.getRule() to not nullable

File: src/main/java/com/exactpro/th2/sim/impl/SimulatorRuleInfo.java [hunk 2]
```diff
@@ -67,6 +69,10 @@ public void handle(Message message) {
         rule.handle(this, message);
     }
 
+    public void touch(@NotNull Map<String, String> args) {
+        rule.touch(this, args);
```
Comment (Nikita-Smirnov-Exactpro) [inline src/main/java/com/exactpro/th2/sim/impl/SimulatorRuleInfo.java:73]: Please add required not null check into the constructor

**Review by Nikita-Smirnov-Exactpro — COMMENTED** [review]
Please add information about the new gRPC into README.

**Review by Nikita-Smirnov-Exactpro — APPROVED** [review]

**Review by jean-michel-perli-sonsin — APPROVED** [review]
Hi, all the text is correct/easy to understand. Thanks!

**Pull request merged**